validator/validator.py

```diff
@@ -0,0 +1,362 @@
+import copy
+import json
+import logging
+import sys
+
+from enum import Enum
+
+import anymarkup
+import click
+import jsonschema
+import requests
+
+logging.basicConfig(format='%(levelname)s: %(message)s', level=logging.INFO)
+
+
+class IncorrectSchema(Exception):
+    def __init__(self, got, expecting):
+        message = "incorrect schema: got `{}`, expecting `{}`".format(
+            got, expecting)
+        super(Exception, self).__init__(message)
+
+
+class MissingSchemaFile(Exception):
+    def __init__(self, path):
+        self.path = path
+        message = "schema not found: `{}`".format(path)
+        super(Exception, self).__init__(message)
+
+
+class ValidatedFileKind(Enum):
+    SCHEMA = "SCHEMA"
+    DATA_FILE = "FILE"
+    REF = "REF"
+
+
+class ValidationResult(object):
```

Comment:
I'm not really sure how I feel about using inheritance here. Especially since the parent is testing to even see if attributes exist on the child.

Might be worth the KISS principle here and raise exceptions for the validation errors that have a method for generating a response. Otherwise just return the Ok dictionary.

There's no reason the summary function has to be a part of a class to be reused.

Reply:
I agree this is a bit messy. I wanted to refactor it, but didn't include in the PR not to overcomplicate it. Will create a task to fix this in the future.